app/auth/auth_service.ts

```diff
@@ -87,6 +95,16 @@ export class AuthService {
     return requestContext;
   }
 
+  setSelectedGroupId(groupId: string) {
+    window.localStorage[SELECTED_GROUP_ID_LOCAL_STORAGE_KEY] = groupId;
+    const selectedGroup = this.user.groups.find((group) => group.id === groupId);
+    if (!selectedGroup) {
+      this.refreshUser();
+    } else {
+      this.emitUser(Object.assign(new User(), this.user, { selectedGroup }));
```

Comment:
`Object.assign` always freaks me out a bit because it copies over all kinds of stuff from the object's prototype as well as the plain values you're expecting it to - just did a small test on copying a proto using Object.assign and it looks slightly different than the original proto

<img width="580" alt="Screen Shot 2020-09-25 at 12 36 13 PM" src="https://user-images.githubusercontent.com/1704556/94308796-e93d3280-ff2b-11ea-8e97-c4525ac7e1c7.png">

i'd just do something simple here like:
```
this.user.selectedGroup = selectedGroup;
this.emitUser(this.user);
```

but i'll leave it up to you